pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -165,7 +165,12 @@ func (c *BootstrapTeardownController) canRemoveEtcdBootstrap() (bool, bool, erro
 		return false, hasBootstrap, nil
 	}
 
-	if len(members) < 4 {
+	isUnsupportedUnsafeEtcd, err := c.isUnsupportedUnsafeEtcd()
+	if err != nil {
+		return false, hasBootstrap, err
+	}
+
+	if !isUnsupportedUnsafeEtcd && len(members) < 4 {
```

Comment:
make this a switch/case.  You missed adding a condition to be sure you had at least two members.

Reply:
done